operationsgateway_api/src/records/waveform.py

```diff
@@ -1,34 +1,47 @@
 import base64
 from io import BytesIO
+import json
 import logging
 
+from botocore.exceptions import ClientError
 import matplotlib.pyplot as plt
 
-from operationsgateway_api.src.exceptions import MissingDocumentError
+from operationsgateway_api.src.exceptions import EchoS3Error, WaveformError
 from operationsgateway_api.src.models import WaveformModel
-from operationsgateway_api.src.mongo.interface import MongoDBInterface
+from operationsgateway_api.src.records.echo_interface import EchoInterface
 
 
 log = logging.getLogger()
 
 
 class Waveform:
+    echo_prefix = "waveforms"
+
     def __init__(self, waveform: WaveformModel) -> None:
         self.waveform = waveform
         self.thumbnail = None
         self.is_stored = False
 
+    def to_json(self):
+        """
+        Use `self.waveform` and return a JSON file stored in a BytesIO object
+        """
+        b = BytesIO()
+        b.write(self.waveform.model_dump_json(indent=2).encode())
+        b.seek(0)
+        return b
+
     async def insert_waveform(self) -> None:
```

Comment:
Is this still meant to be an asynchronus function?

Reply:
Fixed in 861ebc2e4d86e1d6bd48ef1eba48cb77c4a98c7d